payments/management/commands/one_time_sms_message.py

```diff
@@ -0,0 +1,171 @@
+        # - Jollapaikat
+        # - Traileripaikat
+        # - Non-billable customer group (Ei laskutettavat)
+        # - Customers who are manually set to error (you know)
+        # - Vasikkasaaren venesatama (all customers)
+
+        # Exclude jollas and trailer places
+        berth_types = BerthType.objects.exclude(
+            mooring_type__in=(
+                BerthMooringType.TRAWLER_PLACE,
+                BerthMooringType.DINGHY_PLACE,
+            )
+        )
+
+        # Get the ids of the leases, to later get the orders
+        leases = (
+            # Exclude the Vasikkasaari harbor
+            BerthLease.objects.exclude(
+                berth__pier__harbor_id="4e982440-d032-4d9e-a827-e1c89e86da51"
+            )
+            .filter(
+                start_date__year=2021,
+                status=LeaseStatus.OFFERED,
+                berth__berth_type__in=berth_types,
+            )
+            .values("id")
+        )
+        orders = Order.objects.filter(
+            Q(
+                # Valid leases
+                Q(_lease_object_id__in=leases)
+                # Invoices that haven't been paid
+                & Q(status=OrderStatus.WAITING)
+                & Q(
+                    # Either "normal" customers
+                    Q(customer__organization__isnull=True)
+                    # Or "billable" organization customers
+                    | Q(
+                        customer__organization__organization_type__in=(
+                            OrganizationType.COMPANY,
+                            OrganizationType.OTHER,
+                        )
+                    )
+                )
+                # Only for production, since no invoices will have phone, it will work
+                # as a flag to check which ones to send
+                # customer_phone__isnull=True,
+            )
+        )
+
+        profile_service = ProfileService(profile_token=profile_token)
+        sms_service = SMSNotificationService()
+        payment_provider = BamboraPayformProvider(
+            ui_return_url=settings.VENE_UI_RETURN_URL
+        )
+
+        MESSAGE = (
+            "Hyvä asiakas. Tässä maksulinkki venepaikan maksamiseen purjehduskaudelle 2021. "
+            "Maksathan laskun 31.1.2021 mennessä.\n\n"
+            "Dear Customer. Here is a payment link for paying for a berth for the sailing season 2021. "
+            "Please pay the invoice by January 31, 2021.\n\n"
+            "{url}"
+        )
+
+        if dry_run:
+            invoices = orders.count()
+            without_phone = orders.filter(
+                Q(Q(customer_phone__isnull=True) | Q(customer_phone=""))
+            ).count()
+
+            self.stdout.write(f"Invoices to be sent: {invoices}")
+            self.stdout.write(f"Customers without phone: {without_phone}")
+
+            return
+
+        for order in orders:
+            # If the order doesn't have a customer phone number, fetch it from Profiili
```

Comment:
We should try do avoid errors inside the sms sending loop, so that the action doesn't stop halfway due to some error.
For example, is it possible that get_profile() fails, for example some customer doesn't have a profile? Maybe fetch the profiles outside of the loop?

Reply:
Try catch block addresses this.